python/tskit/drawing.py

```diff
@@ -567,6 +581,46 @@ def draw_y_axis(
                         text_anchor="end",
                     )
 
+    def shade_background(
+        self,
+        breaks,
+        tick_length_lower,
+        tree_width=None,
+        bottom_padding=None,
+    ):
+        if tree_width is None:
+            tree_width = self.plotbox.width
+        if bottom_padding is None:
+            bottom_padding = self.plotbox.pad_bottom
+        plot_breaks = self.x_transform(np.array(breaks))
+        dwg = self.drawing
+
+        # For tree sequences, we need to add on the background shaded regions
+        self.root_groups["background"] = self.dwg_base.add(dwg.g(class_="background"))
+        y = self.image_size[1] - self.x_axis_offset
+        for i in range(1, len(breaks)):
+            break_x = plot_breaks[i]
+            prev_break_x = plot_breaks[i - 1]
+            tree_x = i * tree_width + self.plotbox.left
+            prev_tree_x = (i - 1) * tree_width + self.plotbox.left
+            # Shift diagonal lines between tree & axis into the treebox a little
+            diag_height = y - (self.image_size[1] - bottom_padding)
+            self.root_groups["background"].add(
+                dwg.path(
+                    f"M{rnd(prev_tree_x):g},0 "
+                    f"l{rnd(tree_x-prev_tree_x):g},0 "
+                    f"l0,{rnd(y - diag_height):g} "
+                    f"l{rnd(break_x-tree_x):g},{rnd(diag_height):g} "
+                    # NB for curves try "c0,{1} {0},0 {0},{1}" instead of above
```

Comment:
Hmm, I posted a reply to this, but GH seems to have swallowed it. It's to use bezier curves in the background shading, as suggested by @benjeffery (example below). I'm not sure it's an improvement, but probably worth thinking about. I've now changed the way the string is constructed so that the comment is more obvious:

```python
dwg.path(
    "M{start_x:g},0 l{tree_w:g},0 l0,{tree_h:g} "
    "l{rdiag_x:g},{diag_h:g} "
    # or bezier curve equiv "c0,{diag_h} {rdiag_x},0 {rdiag_x},{diag_h}"
    "l0,{tick_h:g} l{ax_x:g},0 l0,-{tick_h:g} "
    "l{ldiag_x:g},-{diag_h:g} "
    # or bezier equiv: "c0,-{diag_h} {ldiag_x},0 {ldiag_x},-{diag_h}"
    "l0,-{tree_h:g}z".format(
        ...
```

<img width="1000" alt="Screenshot 2021-03-23 at 08 53 44" src="https://user-images.githubusercontent.com/4699014/112119449-4b15e680-8bb5-11eb-8bc0-b277c6c185d7.png">

Reply:
That's not obvious to a code reader coming in fresh. This is just confusing I think - if we're not planning on using it then we can delete it. Either that or make it clear that this is an alternative implementation.

Reply:
Perhaps a comment that this is an alternative implementation possibility? I just realised that this isn't part of this PR anyway: it was there previously. But maybe worth fixing with this PR?

Reply:
Right, yes. The code was changed here and in this diff, so let's fix it now. Either fix it or open an issue to track.

Reply:
I've stuck it as a 5 line comment just before the path definition:

```
# NB - the path below draws stright diagonal lines between the tree
# boxes and the X axis. An alternative implementation using bezier
# curves could substitute the following instead of lines 2 and 4 below
# "l0,{box_h:g} c0,{diag_h} {rdiag_x},0 {rdiag_x},{diag_h} "
# "c0,-{diag_h} {ldiag_x},0 {ldiag_x},-{diag_h} l0,-{box_h:g}z"
```